### PR TITLE
Issue/6360 canada fee set manually

### LINKED ISSUE #6360: [Mobile Payments] Charge fees on PaymentIntent creation for IPP payments in Canada
In Canada, we accept Interac payments, which are authorized and captured in a single step on the app.

For most IPP transactions, we rely on the capture step happening on the WCPay Server, so that fees (percentage charge to the merchant) can be added there. For Interac transactions, the only opportunity to add the fees is when we create the `PaymentIntent`, as there is no separate capture step.

We create the `PaymentIntent` using the Stripe Terminal SDK, i.e. from the app not from WCPay.

Since we can't know whether a customer will use Interac or another debit/credit card until after we create a `PaymentIntent`, the workaround we will use for 1.0 is to add the fees to the `PaymentIntent` when we create it, **for IPP transactions in Canada only**.

The fees are as follows:

> 2.6% + $0.25 CAD
– pdfdoF-Gj-p2#comment-1394

For non-Interac transactions, the fees we set in the app will be overridden by WCPay Server at the capture step. We've agreed to still only set these fees in Canada, to reduce any risks to the workflows of the existing merchant base in the US.

This is needed for the initial release of IPP in Canada.

See pdfdoF-Gj-p2#comment-1381 for full discussion

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #6360 

### Description

The PR adds a fee calculation on the client-side. (Canada only) The formula is the following:

```
15 cents flat 
```

### Testing instructions

* Make a payment in the US and check stripe dashboard, notice that the fee is calculated by 2.6% + $0.10
* Make a payment in the CA (interac) and check stripe dashboard, notice that the fee is calculated by $0.15 CA
* Make a payment in the CA (not Interac) and check stripe dashboard, notice that the fee is calculated by 2.6% + $0.10 CA (WCpay overrides the client side value we set)

(still not clear but apparently it has to be `2.7 + 5 cents` but it's on the backed so we are good to proceed here)

~~UPD: added do not merge because still not clear what fee is for the interac payments~~

### CODE REVIEW

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModelTest.kt [hunk 1]
```diff
@@ -1960,6 +1960,57 @@ class CardReaderPaymentViewModelTest : BaseUnitTest() {
             assertThat(captor.firstValue.isPluginCanSendReceipt).isTrue()
         }
 
+    @Test
+    fun `given canada and total 0,58, when flow started, then fee set to 27`() =
```
Comment (rossanafmenezes) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModelTest.kt]: 🔍  Small detail here: the test name is not corresponding to the actual test. It says the fee is set to 27 but the fee is set to 15. 😬

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModelTest.kt [hunk 2]
```diff
@@ -1960,6 +1960,57 @@ class CardReaderPaymentViewModelTest : BaseUnitTest() {
...
+            // When
+            viewModel.start()
+
+            // Then
+            verify(cardReaderManager).collectPayment(captor.capture())
+            assertThat(captor.firstValue.feeAmount).isEqualTo(15)
+        }
+
+    @Test
+    fun `given canada and total 135,6, when flow started, then fee set to 27`() =
```
Comment (rossanafmenezes) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModelTest.kt]: Same as above :)

**Review by rossanafmenezes — COMMENTED** [review]
LGTM, however, quick question about the actual fee amount. The fees do not correspond to what was mentioned in the issue. In the issue, it says 2.6% + $0.25 CAD but here $0.15 CA. I'm probably missing some context here 😬  but I got a bit confused.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModelTest.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModelTest.kt]: Good catch! Fixed - b9369e423fdc3472696182c9e47d4fb0d4516857

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModelTest.kt (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModelTest.kt]: Fixed - b9369e423fdc3472696182c9e47d4fb0d4516857

**Comment by kidinov** [comment]
@rossanafmenezes thank you for the review 🙏

We found out that the fee for Interac is flat 15 cents later than the issue was created. I left a comment [on the issue](https://github.com/woocommerce/woocommerce-android/issues/6360#issuecomment-1113213560)

[Context](p1651045399821679-slack-C025A8VV728)

**Review by rossanafmenezes — APPROVED** [review]
Thanks for the clarification, Andrei! 🥳

**Pull request merged**